src/main/webapp/app/admin/blacklist-agent/blacklisted-agent-management.component.ts

```
@@ -0,0 +1,49 @@
+import { Component, OnInit } from '@angular/core';
+import { BlacklistedAgent, BlacklistedAgentStatus } from './blacklisted-agent';
+import { Observable } from 'rxjs';
+import { BlacklistedAgentService } from './blacklisted-agent.service';
+import { ModalUtils } from '../../shared/index';
+import { BlacklistedAgentChangeStatusDialogComponent } from './blacklisted-agent-change-status-dialog/blacklisted-agent-change-status-dialog.component';
+
+@Component({
+    selector: 'jr2-blacklisted-agent-management',
+    templateUrl: './blacklisted-agent-management.component.html',
+    styleUrls: []
+})
+export class BlacklistedAgentManagementComponent implements OnInit {
+    blacklistedAgents$: Observable<BlacklistedAgent[]>;
+    filterText: string;
+
+    constructor(
+        private blacklistedAgentService: BlacklistedAgentService,
+        private modalUtils: ModalUtils) {
+    }
+
+    ngOnInit(): void {
+        this.getAllBlacklistedAgents();
+    }
+
+    activated(agent: BlacklistedAgent) {
+        return !!agent && agent.status === BlacklistedAgentStatus.ACTIVE;
+    }
+
+    isActive(agent: BlacklistedAgent) {
+        this.changeStatus(agent, BlacklistedAgentStatus.ACTIVE);
+    }
+
+    deactivate(agent: BlacklistedAgent) {
+        this.changeStatus(agent, BlacklistedAgentStatus.INACTIVE);
+    }
+
+    private getAllBlacklistedAgents() {
+        this.blacklistedAgents$ = this.blacklistedAgentService.getAllBlacklistedAgents();
+    }
+
+    private changeStatus(agent: BlacklistedAgent, status: BlacklistedAgentStatus) {
+        this.modalUtils.openLargeModal(BlacklistedAgentChangeStatusDialogComponent)
+            .result
+            .then(() => this.blacklistedAgentService.changeStatus(agent, status)
+                .subscribe(() => this.getAllBlacklistedAgents()))
+            .catch(() => Observable.empty());
```

what will happen in case of an error? it seems that you swallow it

---

there are not any requirements what should happen then; error will not be processed

---

throw it into the console at least or show something... we shouldn't hide errors as it will lead to debugging problems

---

changed